src/base/BaseActionsRouter.sol

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.24;
+
+import {IPoolManager} from "@uniswap/v4-core/src/interfaces/IPoolManager.sol";
+import {Currency} from "@uniswap/v4-core/src/types/Currency.sol";
+import {TransientStateLibrary} from "@uniswap/v4-core/src/libraries/TransientStateLibrary.sol";
+
+import {SafeTransferLib} from "solmate/utils/SafeTransferLib.sol";
+
+import {SafeCallback} from "./SafeCallback.sol";
+import {ReentrancyLock} from "./ReentrancyLock.sol";
+import {Actions} from "../libraries/Actions.sol";
+import {Locker} from "../libraries/Locker.sol";
+import {BytesLib} from "../libraries/BytesLib.sol";
+
+/// @notice Abstract contract for performing a combination of actions on Uniswap v4.
+/// @dev If an inheriting contract does not want to support all commands, they should simply revert with the
+/// UnsupportedAction error in their overriding function
+/// The _msgSender function returns the msg.sender from the call to executeActions. This allows actions to know which
+/// address instructed the action to be requested. The msg.sender within the unlockCallback with be the v4 PoolManager.
+abstract contract BaseActionsRouter is SafeCallback, ReentrancyLock {
+    using TransientStateLibrary for IPoolManager;
+    using SafeTransferLib for address;
+    using BytesLib for bytes;
+
+    /// @notice emitted when different numbers of parameters and actions are provided
+    error LengthMismatch();
+
+    /// @notice emitted when an inheriting contract does not support an action
+    error UnsupportedAction(uint256 action);
+
+    constructor(IPoolManager poolManager) SafeCallback(poolManager) {}
+
+    /// @notice internal function that triggers the execution of a set of actions on v4
+    /// TODO move the reentrancy lock modifier up a level
```

Comment:
to where?

Reply:
I'm removing it from this contract. UR already has a reentrancy lock because it needs one wider across all commands. So I think we should just have the upper level contracts implement them (if they need them).